core/src/test/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -316,6 +316,44 @@ public static double randomDouble() {
         return random().nextDouble();
     }
 
+    /**
+     * Returns a double value in the interval [start, end) if lowerInclusive is set to true,
+     * (start, end) otherwise.
+     *
+     * @param start lower bound of interval to draw uniformly distributed random numbers from
+     * @param end upper bound
+     * @param lowerInclusive whether or not to include lower end of the interval
+     * */
+    public static double randomDoubleBetween(double start, double end, boolean lowerInclusive) {
+        double result = 0.0;
+
+        if (start == -Double.MAX_VALUE || end == Double.MAX_VALUE) {
+            // formula below does not work with very large doubles
+            result = Double.longBitsToDouble(randomLong());
+            while(result < start || result > end || Double.isNaN(result)) {
+                result = Double.longBitsToDouble(randomLong());
+            }
+        } else {
+            result = randomDouble();
+            if (lowerInclusive == false) {
+                while (result <= 0.0) {
+                    result = randomDouble();
+                }
+            }
+            result = result * end + (1.0 - result) * start;
+        }
+        return result;
+    }
+
+    /** Randomly generate either NaN or one of the two infinity values. */
+    public static double invalidRandomDouble() {
```

Comment:
actually, I see it in your test too but the content is different, not sure which one of the two is more correct, but I'd leave the method there rather than here.